sdk/nodejs/runtime/rpc.ts

```diff
@@ -190,7 +190,7 @@ export function resolveProperties(
             else {
                 // We're previewing. If the engine was able to give us a reasonable value back,
                 // then use it. Otherwise, inform the Output that the value isn't known.
-                const isKnown = value !== undefined;
+                const isKnown = !containsUnknowns(value);
```

Comment:
should we do this unilaterally in update as well?

should we do the reverse?  i.e. should update strongly assert that it doesn't not contain unknowns?

Reply:
as per comment below... it seems unfortunate that this can't be done inside the Output type itself so that the callers don't need to massage like this.  i.e. since the Output is getting that value that contains unknowns or not, it can set its own isKnown bit properly, right?

or is there some subtlety here i'm missin.

Reply:
We will in fact recalculate this inside the `Output` constructor, but that constructor is hidden behind this layer of resolvers. We could just pass `true` here, tbf.

Reply:
that would be my preference.  indeed, if 'unknown' can be computed *entire* from the input value, i would love it if we could just move to that and have all external code just be blisfully unware of this concept entirely.